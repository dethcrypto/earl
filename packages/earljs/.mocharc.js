process.env.NODE_ENV = 'test'
module.exports = {
  spec: '{src,test}/**/*.test.ts',
  file: 'test/setup.ts',
  require: 'ts-node/register/transpile-only',
  watchExtensions: 'ts',
  extension: 'ts',
}
