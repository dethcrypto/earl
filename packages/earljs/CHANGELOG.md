# earljs

## 0.1.5

### Patch Changes

- 1eac490: Add `toMatchSnapshot` validator

## 0.1.4

### Patch Changes

- 591dccc: Add readme to the final npm package

## 0.1.3

### Patch Changes

- 4964a16: Migrated to monorepo.
- 4964a16: Fix bug in `StringMatching` matcher that prevented it from matching strings with special characters
- 1e3951d: Added new matchers: `arrayWith`, `objectWith` Improve `containerWith` to work with multiple values
- 4964a16: Added experimental plugin API.

## 0.1.2

### Patch Changes

- 0f490bd: Tweak `toBeRejected` validator signature to be properly async (return `Promise<void>`). This was a bug in
  typings not in behaviour.

## 0.1.1

### Patch Changes

- d2efbc6: Change compilation target to ES2015 to enable running on older node.js versions
- 42ed317: Added `containerWith` matcher, allowing to match a iterable containing given value
