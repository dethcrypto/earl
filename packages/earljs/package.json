{
  "name": "earljs",
  "description": "Ergonomic, modern and type-safe assertion library",
  "keywords": [
    "assert",
    "assertion",
    "assertions",
    "modern",
    "mocha",
    "chai",
    "jest",
    "typescript",
    "test",
    "testing",
    "unit",
    "tdd",
    "bdd",
    "expect",
    "exception"
  ],
  "repository": "dethcrypto/earl",
  "author": "Kris Kaczor <chris@kaczor.io>",
  "license": "MIT",
  "version": "0.2.3",
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "files": [
    "dist/**/*",
    "internals/package.json",
    "mocha/package.json",
    "uvu/package.json"
  ],
  "scripts": {
    "start": "ts-node -T ./src/index.ts",
    "format": "prettier --ignore-path ../../.prettierignore --check .",
    "format:fix": "prettier --ignore-path ../../.prettierignore --write .",
    "lint": "eslint --ext .ts src test",
    "lint:fix": "pnpm lint --fix",
    "typecheck": "tsc --noEmit",
    "clean": "rimraf dist",
    "build": "tsc -p ./tsconfig.prod.json",
    "build:watch": "pnpm clean && tsc -p ./tsconfig.prod.json --watch",
    "build:watch:test": "pnpm clean && tsc -p ./tsconfig.json --watch",
    "test": "mocha",
    "test:fix": "pnpm lint:fix && pnpm format:fix && pnpm typecheck && pnpm test",
    "release": "pnpm lint && pnpm format && pnpm typecheck && pnpm test && pnpm build"
  },
  "dependencies": {
    "debug": "^4.1.1",
    "jest-snapshot": "^26.6.2",
    "ts-essentials": "^6.0.5"
  },
  "devDependencies": {
    "@microsoft/tsdoc": "^0.13.0",
    "@types/debug": "^4.1.5",
    "error-stack-parser": "^2.0.6",
    "rimraf": "^3.0.2",
    "typescript": "^4.6"
  }
}
