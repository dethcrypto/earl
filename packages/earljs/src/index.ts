export { Control, ValidationResult } from './Control'
export * from './expect'
export * from './format'
export * from './isEqual'
import './validators/toEqual'
import './validators/snapshots/toMatchSnapshot'
import './validators/snapshots/TestContext'
import './matchers/a'
import './matchers/anything'
import './matchers/arrayOfLength'
import './matchers/stringMatching'
import './matchers/nullish'
