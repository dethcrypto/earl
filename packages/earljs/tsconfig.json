{
  "extends": "../../tsconfig.options.json",
  "include": ["src", "test", "generate-docs.ts"]
}
