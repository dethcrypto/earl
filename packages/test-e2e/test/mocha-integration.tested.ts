import { expect } from 'earljs'

describe('earljs/mocha', () => {
  it('successful test', () => {
    expect(2).toMatchSnapshot()
  })
  it('failing test', () => {
    expect(Math.random()).toMatchSnapshot()
  })
})
