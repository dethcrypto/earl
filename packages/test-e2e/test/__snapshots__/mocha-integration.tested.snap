// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`earljs/mocha failing test 1`] = `0.3034235109109409`;

exports[`earljs/mocha successful test 1`] = `2`;

exports[`earljs/mocha successful test 2`] = `2`;
